src/elements/core/mixins/form-associated-radio-button-mixin.ts

```diff
@@ -37,56 +37,74 @@ export declare class SbbFormAssociatedRadioButtonMixinType
   protected navigateByKeyboard(radio: SbbFormAssociatedRadioButtonMixinType): Promise<void>;
 }
 
+type RadioButtonGroup = {
+  name: string;
+  form: HTMLFormElement | null;
+  radios: SbbFormAssociatedRadioButtonMixinType[];
+};
+
 /**
- * A static registry that holds a collection of `radio-buttons`, grouped by `name`.
+ * A static registry that holds a collection of `radio-buttons`, grouped by `name + form`.
  * It is mainly used to support the standalone groups of radios.
+ * The identifier of a group is composed of the couple 'name + form' because multiple radios with the same name can coexist (as long as they belong to different forms)
  * @internal
  */
 export class RadioButtonRegistry {
-  private static _registry: { [x: string]: SbbFormAssociatedRadioButtonMixinType[] } = {};
+  private static _registry: RadioButtonGroup[] = [];
 
   private constructor() {}
 
   /**
-   * Adds @radio to the @groupName group. Checks for duplicates
+   * Adds @radio to the '@groupName + @form' group. Checks for duplicates
```

Comment:
It's probably too sophisticated: But how does native behave vs. our implementation when there is a radio in a shadow DOM (like visual regression app) with the same name as in light DOM. In our case there would be the same group, but native probably not? But as said, not sure whether we have to support it :D It would be easy solvable by using a distinct name.